packages/@aws-cdk/aws-s3-assets/lib/asset.ts

```diff
@@ -148,9 +146,13 @@ export class Asset extends cdk.Construct implements cdk.IAsset {
     this.s3ObjectUrl = location.s3ObjectUrl;
     this.httpUrl = location.httpUrl;
     this.s3Url = location.httpUrl; // for backwards compatibility
-    this.kmsKey = location.kmsKeyArn ? kms.Key.fromKeyArn(scope, 'Key', location.kmsKeyArn) : undefined;
 
-    this.bucket = s3.Bucket.fromBucketName(this, 'AssetBucket', this.s3BucketName);
+    const kmsKey = location.kmsKeyArn ? kms.Key.fromKeyArn(scope, 'Key', location.kmsKeyArn) : undefined;
```

Comment:
Do not pollute the parent scope:

```suggestion
    const kmsKey = location.kmsKeyArn ? kms.Key.fromKeyArn(this, 'Key', location.kmsKeyArn) : undefined;
```

Reply:
Ah right, good catch. Sorry about that.